lib/core/msdf-sys/src/lib.rs

```diff
@@ -0,0 +1,184 @@
+                "msdfgen_loadFontMemory",
+                emscripten_data_types::NUMBER,
+                param_types,
+                params);
+            Font { handle }
+        }
+    }
+}
+
+/*
+ * Note [Usage of Uint8Array::view]
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ * We use view in this place to avoid copying font data. This is the only way
+ * to do it with js_sys structures. The Uint8Array does not leave function
+ * scope, so does not excess lifetime of data
+ */
+
+impl Drop for Font {
+    fn drop(&mut self) {
+        msdfgen_free_font(self.handle.clone())
+    }
+}
+
+// =======================================
+// === MutlichannelSignedDistanceField ===
+// =======================================
+
+// Mutlichannel Signed Distance Field for one glyph
+//
+// For more information about MSDF see https://github.com/Chlumsky/msdfgen
+pub struct MutlichannelSignedDistanceField {
+    pub width  : usize,
+    pub height : usize,
+    pub data   : [f32; Self::MAX_DATA_SIZE]
+}
+
+// Parameters of MSDF generation
+//
+// The structure gathering MSDF generation parameters meant to be same for all rendered glyphs
+pub struct MSDFParameters {
+    pub width                         : usize,
+    pub height                        : usize,
+    pub edge_coloring_angle_threshold : f64,
+    pub range                         : f64,
+    pub edge_threshold                : f64,
+    pub overlap_support               : bool
+}
+
+impl MutlichannelSignedDistanceField {
+    pub const MAX_SIZE       : usize = 32;
+    pub const CHANNELS_COUNT : usize = 3;
+    pub const MAX_DATA_SIZE  : usize = Self::MAX_SIZE * Self::MAX_SIZE *
+        Self::CHANNELS_COUNT;
+
+    pub fn generate(
+        font      : &Font,
+        unicode   : u32,
+        params    : MSDFParameters,
+        scale     : Vector2D<f64>,
+        translate : Vector2D<f64>,
+    ) -> MutlichannelSignedDistanceField {
+        assert!(params.width  <= Self::MAX_SIZE);
+        assert!(params.height <= Self::MAX_SIZE);
+
+        let output_address = msdfgen_generate_msdf(
+            params.width,
+            params.height,
+            font.handle.clone(),
+            unicode,
+            params.edge_coloring_angle_threshold,
+            params.range,
+            scale.x,
+            scale.y,
+            translate.x,
+            translate.y,
+            params.edge_threshold,
+            params.overlap_support
+        );
+        let mut data : [f32; Self::MAX_DATA_SIZE] = [0.0; Self::MAX_DATA_SIZE];
+        let data_size = params.width*params.height*Self::CHANNELS_COUNT;
+        copy_f32_data_from_msdfgen_memory(
+            output_address,
+            &mut data,
+            data_size
+        );
+        MutlichannelSignedDistanceField {
+            width: params.width,
+            height: params.height,
+            data
+        }
+    }
+}
+
+
+
+#[cfg(test)]
+mod tests {
+    extern crate wasm_bindgen_test;
+    extern crate slice_as_array;
```

Comment:
Do we need to use `extern crate` in Rust 2018? I believe its not needed anymore

Reply:
It isn't. Removed